subo/command/create_handler.go

```diff
@@ -0,0 +1,71 @@
+package command
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/suborbital/atmo/directive"
+	"github.com/suborbital/subo/builder/context"
+	"github.com/suborbital/subo/subo/util"
+)
+
+type handlerData struct {
+	HandlerType string
+	Request     string
+	Method      string
+}
+
+func CreateHandlerCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "handler <resource>",
+		Short: "create a new handler",
+		Long:  `create a new handler for Subo CLI`,
+		Args:  cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			resource := args[0]
+
+			handlerType, _ := cmd.Flags().GetString(typeFlag)
+			method, _ := cmd.Flags().GetString(methodFlag)
+
+			dir, _ := cmd.Flags().GetString(dirFlag)
+
+			util.LogStart(fmt.Sprintf("creating handler with function name %s", resource))
+
+			bctx, err := context.ForDirectory(dir)
+			if err != nil {
+				return errors.Wrap(err, "failed to ForDirectory")
+			}
+
+			if bctx.Directive == nil {
+				util.LogFail("Handlers can only be created in projects with a Directive.yaml file")
+				return errors.New("Directive.yaml not found")
+			}
+			//Create a new handler object
+			handler := directive.Handler{
+				Input: directive.Input{
+					Type:     handlerType,
+					Resource: resource,
+					Method:   method,
+				},
+			}
+
+			//Add the handler object to the directive file
+			bctx.Directive.Handlers = append(bctx.Directive.Handlers, handler)
+
+			//Write Directive File which overwrites the entire file
```

Comment:
Before writing the file, I would call `bctx.Directive.Validate()` and fail if there are any errors

Reply:
We decided against this and will think about implementing on second interation: with steps[].